server/website/website/models.py

```diff
@@ -182,17 +182,22 @@ class Meta:  # pylint: disable=no-init
 
 class SessionKnobManager(models.Manager):
     @staticmethod
-    def get_knobs_for_session(session):
+    def get_knobs_for_session(session, only_session_knobs=False):
```

Comment:
Instead of adding an extra arg here and in async_tasks.py can you add a function in views.py that sets the session knobs like the management command that Dongsheng added? https://github.com/cmu-db/ottertune/blob/master/server/website/website/management/commands/loadknob.py

Reply:
Sure, Dana. I can set the session knobs in the test. However, current version gets min/max from the session knobs, and then gets min/max from knob fixture.  https://github.com/bohanjason/ottertune/blob/master/server/website/website/fixtures/postgres-96_knobs.json   The knob fixture may not have correct min/max since there are hundreds of knobs in the fixture.  I run into such issues multiple times.

I am thinking of using min/max in the session knobs, and using min/max in the training data for other knobs. Which one you prefer?

Reply:
I would explicitly set the knobs you want to tune and their minvals/maxvals instead of relying on the minvals/maxvals from the session knobs and the knob fixture. You could store them in a JSON file along with your other test files.

Since we'll be running this test on travis I would also set the minval/maxval of memory (and possibly CPU) knobs conservatively so that we do not exceed the resources available on the VM. For example, the default value of shared_buffers is 128 MB so you might set minval to 64 MB and maxval to 256 MB.

Do you agree that this would be better than using the values from the session knobs and knob fixture? If not let me know.

Reply:
Sounds good. I will look into this. Thanks, Dana